ninshiki_py/detector/tflite.py

```diff
@@ -0,0 +1,132 @@
+        self.floating_model = None
+        self.input_mean = 0
+        self.input_std = 0
+
+        self.initiate_tflite()
+
+    def initiate_tflite(self):
+        self.interpreter = Interpreter(model_path=self.model)
+        self.interpreter.allocate_tensors()
+
+        # Get model details
+        self.input_details = self.interpreter.get_input_details()
+        self.output_details = self.interpreter.get_output_details()
+
+        if len(self.input_details) >= 1:
+            if 'shape' in self.input_details[0].keys():
+                self.model_height = self.input_details[0]['shape'][1]
+                self.model_width = self.input_details[0]['shape'][2]
+
+        self.floating_model = (self.input_details[0]['dtype'] == np.float32)
+
+        self.load_metadata()
+
+    def load_metadata(self):
+        # Load metadata from model.
+        displayer = metadata.MetadataDisplayer.with_model_file(self.model)
+
+        # Save model metadata for preprocessing later.
+        model_metadata = json.loads(displayer.get_metadata_json())
+
+        mean = 127.5
+        std = 127.5
+        if len(model_metadata) > 1:
+            if 'subgraph_metadata' in model_metadata.keys():
+                process_units = model_metadata['subgraph_metadata'][0][
+                    'input_tensor_metadata'][0]['process_units']
+                for option in process_units:
+                    if option['options_type'] == 'NormalizationOptions':
+                        mean = option['options']['mean'][0]
+                        std = option['options']['std'][0]
+        self.input_mean = mean
+        self.input_std = std
+
+        # Load label list from metadata.
+        file_name = displayer.get_packed_associated_file_list()[0]
+        label_map_file = displayer.get_associated_file_buffer(file_name).decode()
+        label_list = list(filter(len, label_map_file.splitlines()))
+        self.labels = label_list
+
+    def detection(self, image: np.ndarray, detection_result: MsgType, conf_threshold: float):
+        min_conf_threshold = conf_threshold
+        img_height, img_width, _ = image.shape
+
+        # Preprocess
+        image_resized = cv2.resize(image, (self.model_width, self.model_height))
+        input_data = np.expand_dims(image_resized, axis=0)
+        # Normalize pixel values if using a floating model (i.e. if model is non-quantized)
+        if self.floating_model:
+            input_data = (np.float32(input_data) - self.input_mean) / self.input_std
+
+        # Perform the actual detection by running the model with the image as input
+        self.interpreter.set_tensor(self.input_details[0]['index'], input_data)
+        self.interpreter.invoke()
+
+        # Get detection results
+        boxes = self.interpreter.get_tensor(self.output_details[0]['index'])[0]
+        classes = self.interpreter.get_tensor(self.output_details[1]['index'])[0]
+        scores = self.interpreter.get_tensor(self.output_details[2]['index'])[0]
```

Comment:
should we add if guard here? see other lines too.

Reply:
done